x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -104,6 +127,53 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
     return location;
   };
 
+  const getAnomalySourceMapsLink = async (
+    anomaly: AnomaliesTableRecord,
+    sourceIndicesWithGeoFields: SourceIndicesWithGeoFields
+  ) => {
+    // Create a layer for each of the geoFields
+    const initialLayers = getInitialSourceIndexFieldLayers(
+      sourceIndicesWithGeoFields[anomaly.jobId]
+    );
+    // Widen the timerange by one bucket span on start/end to increase chances of always having data on the map
+    const anomalyBucketStartMoment = moment(anomaly.source.timestamp).tz(getDateFormatTz());
+    const anomalyBucketStart = anomalyBucketStartMoment
+      .subtract(anomaly.source.bucket_span, 'seconds')
+      .toISOString();
+    const anomalyBucketEnd = anomalyBucketStartMoment
+      .add(anomaly.source.bucket_span * 3, 'seconds')
+      .toISOString();
```

Comment:
Nit - for consistency with the link to Discover, you could subtract `1s` from the end time.

Reply:
Updated in 8230df0139f4e9520b0370b7d8cdd6ed32acdcfd